### PR TITLE
Balder/handle cased streaming search

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR

### CODE REVIEW

File: streamingvisitors/src/vespa/vsm/searcher/utf8stringfieldsearcherbase.h [hunk 1]
```diff
@@ -53,14 +54,81 @@ class UTF8StringFieldSearcherBase : public StrChrFieldSearcher
...
     };
 
 protected:
     SharedSearcherBuf _buf;
 
-    const search::byte * tokenize(const search::byte * buf, size_t maxSz, cmptype_t * dstbuf, size_t & tokenlen);
+    using byte = search::byte;
+    using Normalizing = search::streaming::Normalizing;
+
+    class TokenizeReader {
```
Comment (geirst) [inline streamingvisitors/src/vespa/vsm/searcher/utf8stringfieldsearcherbase.h:66]: Perhaps this should be moved to a separate .h and .cpp file with its own basic unit test?

**Review by geirst — DISMISSED** [review]
👍

File: config-model/src/main/java/com/yahoo/schema/derived/IndexInfo.java [hunk 2]
```diff
@@ -525,54 +525,33 @@ protected String getDerivedName() {
...
-
-        public int hashCode() {
-            return index.hashCode() + 17 * command.hashCode();
+            return command.isEmpty();
         }
 
         public boolean equals(Object object) {
-            if (!(object instanceof IndexCommand other)) {
-                return false;
+                if (!(object instanceof IndexCommand other)) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/schema/derived/IndexInfo.java:541]: Fixed

File: config-model/src/main/java/com/yahoo/schema/derived/IndexInfo.java [hunk 3]
```diff
@@ -525,54 +525,33 @@ protected String getDerivedName() {
...
+                }
+
+                return other.index.equals(this.index) &&
+                        other.command.equals(this.command);
             }
 
-            return other.index.equals(this.index) &&
-                   other.command.equals(this.command);
-        }
+            public String toString() {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/schema/derived/IndexInfo.java:549]: Fixed

File: config-model/src/test/java/com/yahoo/schema/derived/VsmFieldsTestCase.java [hunk 4]
```diff
@@ -19,24 +23,76 @@
...
 
         assertEquals(1, cfg.fieldspec().size());
         VsmfieldsConfig.Fieldspec fieldSpec = cfg.fieldspec().get(0);
         assertEquals("ref_field", fieldSpec.name());
         assertEquals(VsmfieldsConfig.Fieldspec.Searchmethod.NONE, fieldSpec.searchmethod());
     }
+
+    @Test
+    void test_exact_string() {
+        Schema schema = createSchema();
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config-model/src/test/java/com/yahoo/schema/derived/VsmFieldsTestCase.java]: Fixed

**Review by geirst — APPROVED** [review]
👍

File: streamingvisitors/src/vespa/vsm/searcher/utf8stringfieldsearcherbase.h (hunk 1 above)
Comment (baldersheim) [reply streamingvisitors/src/vespa/vsm/searcher/utf8stringfieldsearcherbase.h:66]: Yes, I will do that in a separate PR.

**Pull request merged**